runhouse/main.py

```diff
@@ -590,155 +592,129 @@ def status(
     _print_status(cluster_status, current_cluster)
 
 
-def get_clusters_from_den():
-    get_clusters_params = {"resource_type": "cluster", "folder": rns_client.username}
-    clusters_in_den_resp = rns_client.session.get(
-        f"{rns_client.api_server_url}/resource",
-        params=get_clusters_params,
-        headers=rns_client.request_headers(),
+def _create_output_table(
+    total_clusters: int,
+    running_clusters: int,
+    displayed_clusters: int,
+    filters_requested: bool,
+):
+    displayed_running_clusters = (
+        running_clusters
+        if running_clusters < displayed_clusters
+        else displayed_clusters
     )
+    table_title = f"[bold cyan]Clusters for {rns_client.username} (Running: {displayed_running_clusters}/{running_clusters}, Total Displayed: {displayed_clusters}/{total_clusters})[/bold cyan]"
 
-    return clusters_in_den_resp
+    table = Table(title=table_title)
 
+    if not filters_requested:
+        table.caption = f"[reset]Showing clusters that were active in the last {int(LAST_ACTIVE_AT_TIMEFRAME / HOUR)} hours."
+        table.caption_justify = "left"
 
-@cluster_app.command("list")
-def cluster_list():
-    """List all Runhouse clusters saved in Den"""
-    import sky
+    # Add columns to the table
+    table.add_column("Name", justify="left", no_wrap=True)
+    table.add_column("Cluster Type", justify="left", no_wrap=True)
+    table.add_column("Status", justify="left")
+    table.add_column("Last Active (UTC)", justify="left")
 
-    # logged out case
-    if not rh.configs.token:
-        # TODO [SB]: adjust msg formatting (coloring etc)
-        sky_cli_command_formatted = f"{italic_bold_ansi}sky status -r{reset_format}"  # will be printed bold and italic
-        console.print(
-            f"Runhouse token required to view all clusters. Please run `runhouse login` to load your token. To view on-demand clusters run {sky_cli_command_formatted}"
-        )
-        return
+    return table
 
-    sky_clusters = sky.status()
 
-    clusters_in_den_resp = get_clusters_from_den()
+def _add_cluster_as_table_row(table: Table, rh_cluster: dict):
+    last_active = rh_cluster.get("Last Active (UTC)")
+    last_active = last_active if last_active != "1970-01-01 00:00:00" else "Unknown"
+    table.add_row(
+        rh_cluster.get("Name"),
+        rh_cluster.get("Cluster Type"),
+        rh_cluster.get("Status"),
+        last_active,
+    )
 
-    if clusters_in_den_resp.status_code != 200:
-        logger.error(
-            f"Failed to load clusters from Den for username: {rns_client.username}"
-        )
-        clusters_in_den = []
-    else:
-        clusters_in_den = clusters_in_den_resp.json().get("data")
+    return table
 
-    clusters_in_den_names = [cluster.get("name") for cluster in clusters_in_den]
 
-    if not sky_clusters and not clusters_in_den:
-        console.print("No existing clusters.")
+def _add_clusters_to_output_table(table: Table, clusters: list[Dict]):
```

Comment:
dumb nit, but since we're using `typing` here lets keep it consistent
```suggestion
def _add_clusters_to_output_table(table: Table, clusters: List[Dict]):
```